### PR TITLE
Expose ondemand feature

### PR DESCRIPTION
**Pull request by ibab**
This exposes the TRACY_ON_DEMAND feature (don't record traces until there is a connection, and allow re-connecting to a client).
I had to make a small change to `make_sys.sh` to make it work on MacOS.

### CODE REVIEW

File: tracy-client/Cargo.toml [hunk 1]
```diff
@@ -21,3 +21,4 @@ enable = [ "tracy-client-sys/enable" ]
 delayed-init = [ "tracy-client-sys/delayed-init" ]
 lowres-timer = [ "tracy-client-sys/lowres-timer" ]
 noexit = [ "tracy-client-sys/noexit" ]
+ondemand = [ "tracy-client-sys/ondemand" ]
```
Comment (nagisa) [inline tracy-client/Cargo.toml:24]: Will want to add a similar one to `tracing-tracy`.

**Comment by nagisa** [comment]
Note that the `make_sys` is still being developed (while I have time) so it is likely that it won't work on e.g. macOS. It is intended to be only run by the GHA, so the changes there aren't necessary. Would be great if this PR only restricted itself to just the `on_demand` feature.

**Comment by nagisa** [comment]
I went ahead and merged the two commits adding the feature manually. Thanks!

**Comment by ibab** [comment]
Thank you!

**Pull request closed without merging**